runtime/src/util/events.rs

```diff
@@ -0,0 +1,128 @@
+use crate::cbor::serialize_vec;
+use crate::ActorError;
+use fvm_shared::event::{ActorEvent, Entry, Flags};
+use serde::ser;
+
+// Codec identifier for CBOR-encoded data.
+const IPLD_CBOR: u64 = 0x51;
+
+/// Builder for ActorEvent objects, accumulating key/value pairs.
+pub struct EventBuilder {
+    entries: Vec<Entry>,
+}
+
+impl EventBuilder {
+    /// Creates a new builder with no values.
+    pub fn new() -> Self {
+        Self { entries: Vec::new() }
+    }
+
+    /// Pushes an entry with an indexed key and no value.
+    pub fn label(mut self, name: &str) -> Self {
```

Comment:
Bikeshedding this convention... it may be better to just define a common key ("label", "message", "type", etc.) and use that. Looking for the "valueless key" could be a bit annoying.

Reply:
I took this pattern from examples in FIP-0049, though it does have examples both ways. My interpretation was that it's the first position that identified the first field as a discriminator of the schema for remaining fields.

"Label" in this utility class is just a word to mean "indexed key with no value". It's not supposed to mean event type. An event could have multiple labels. I'm open to alternative words here.

Discussion about whether we should actually use key-with-no-value in the events here belongs where this is called, and we could switch to `value_indexed` with some new key instead.

But this bikeshedding is what I'm here for!